### PR TITLE
chore: Consolidate typescript to monorepo package.json

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR removes individually declared typescript dependencies and consolidates it at the root level of the monorepo.

#### Description of how you validated changes
`yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -67,9 +67,7 @@
 		"@aws-amplify/react-native": "^1.0.0",
 		"@types/js-cookie": "3.0.2",
 		"@types/uuid": "^9.0.0",
-		"find": "^0.2.7",
```
Comment (Samaritan1011001) [inline packages/core/package.json:70]: just making sure, is this not needed?

File: packages/pubsub/package.json [hunk 2]
```diff
@@ -83,9 +83,7 @@
 		"@aws-amplify/core": "^6.0.0"
 	},
 	"devDependencies": {
-		"@aws-amplify/core": "6.0.0",
-		"cpx": "^1.5.0",
```
Comment (Samaritan1011001) [inline packages/pubsub/package.json:87]: same question here

File: packages/storage/package.json [hunk 3]
```diff
@@ -101,9 +101,7 @@
 	},
 	"devDependencies": {
 		"@aws-amplify/core": "6.0.0",
-		"@aws-amplify/react-native": "^1.0.0",
-		"@types/sinon": "^7.5.1",
```
Comment (Samaritan1011001) [inline packages/storage/package.json:105]: here too

**Review by jimblanc — APPROVED** [review]
Thanks!

File: packages/core/package.json (hunk 1 above)
Comment (cshfang) [reply packages/core/package.json:70]: Yeah even in the PR it was added (originally bootstrapping core) it was never used.

File: packages/pubsub/package.json (hunk 2 above)
Comment (cshfang) [reply packages/pubsub/package.json:87]: I looked up `cpx` usage and searched the repo for cpx and found nothing

File: packages/storage/package.json (hunk 3 above)
Comment (cshfang) [reply packages/storage/package.json:105]: Yep, this is another relic from using `sinon` a long time ago

**Pull request closed without merging**